subsys/bluetooth/audio/codec.c

```diff
@@ -79,8 +158,99 @@ uint8_t bt_audio_codec_cfg_get_val(const struct bt_audio_codec_cfg *codec_cfg, u
 	return param.data_len;
 }
 
+int bt_audio_codec_cfg_set_val(struct bt_audio_codec_cfg *codec_cfg, uint8_t type,
+			       const uint8_t *data, size_t data_len)
+{
+	CHECKIF(codec_cfg == NULL) {
+		LOG_DBG("codec_cfg is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data == NULL) {
+		LOG_DBG("data is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data_len == 0U) {
+		LOG_DBG("data_len is 0");
+		return -EINVAL;
+	}
+
+	for (uint16_t i = 0U; i < codec_cfg->data_len;) {
+		const uint8_t len = codec_cfg->data[i++];
+		const uint8_t data_type = codec_cfg->data[i++];
+		const uint8_t value_len = len - sizeof(data_type);
+
+		if (data_type == type) {
+			uint8_t *value = &codec_cfg->data[i];
+
+			if (data_len == value_len) {
+				memcpy(value, data, data_len);
```

Comment:
when did we stop have `(void)memcpy...` style?

Reply:
Not quite sure. I stopped asking for it in reviews as I was the only one left asking for it, and I don't think our styleguide mentions this anywhere.

Reply:
no worries.